src/main/groovy/org/gradle/android/workarounds/AbstractAbsolutePathWorkaround.groovy

```diff
@@ -0,0 +1,35 @@
+package org.gradle.android.workarounds
+
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.file.FileCollection
+import org.gradle.api.tasks.PathSensitivity
+
+abstract class AbstractAbsolutePathWorkaround implements Workaround {
+    abstract Class<? extends Task> getAndroidTaskClass()
+    abstract String getPropertyName()
+    abstract void setPropertyValue(Task task, FileCollection fileCollection)
+
+    @Override
+    void apply(WorkaroundContext context) {
+        Project project = context.project
+        project.tasks.withType(androidTaskClass).configureEach { Task task ->
+            FileCollection originalPropertyValue
+            // Create a synthetic input with the original property value and RELATIVE path sensitivity
+            project.gradle.taskGraph.beforeTask {
+                if (it == task) {
+                    originalPropertyValue = task.getProperty(propertyName)
+                    setPropertyValue(task, project.files())
+                    task.inputs.files(originalPropertyValue)
+                        .withPathSensitivity(PathSensitivity.RELATIVE)
+                        .withPropertyName("${propertyName}.workaround")
+                        .skipWhenEmpty(true)
+                }
+            }
+            // Set the task property back to its original value
+            task.doFirst {
+                setPropertyValue(task, originalPropertyValue)
+            }
```

Comment:
Does this work with Gradle 6.4+? IIUC the instant execution team is adding some safe-guards against tasks changing their input properties, and with build cache we are going to add more soon.

Reply:
Seems to work with the latest 6.4 nightly, but I wouldn't be surprised if this was locked down at some point for good reason.  It makes me feel dirty.